src/test/outmdsd/MockServer.cc

```
@@ -197,6 +197,30 @@ MockServer::RunEchoTask(
     m_connfdSet.insert(connfd);
     lck.unlock();
 
+    // Because RunEchoTask is called in an async task, it is better to handle
+    // exceptions at task level. Otherwise, the exceptions won't show up in
+    // master thread until task.get() is called, which could cause unexpected delays
+    // at handling client communications.
+    try {
+        StartReadLoop(connfd);
+    }
+    catch(const std::exception& ex) {
+        Log(std::string("Error: RunEchoTask() failed: ") + ex.what());
+
+        std::unique_lock<std::mutex> lck(m_connMutex);
+        m_connfdSet.erase(connfd);
+        lck.unlock();
+
+        shutdown(connfd, SHUT_RDWR);
+        close(connfd);
```

Should you re-throw the exception so that the master thread sees it, too?